### PR TITLE
Upgrade curl to 8.5.0; Addresses CVE-2023-46219

### PR DESCRIPTION
**Pull request by Camelron**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 
Cherry-picked PR https://github.com/microsoft/CBL-Mariner/pull/7059

[AUTOPATCHER-CORE] Upgrade curl to 8.5.0 CVE-2023-46219
Upgrade pipeline run -> https://dev.azure.com/mariner-org/mariner/_build/results?buildId=471532&view=results

buddy build -> https://dev.azure.com/mariner-org/mariner/_build/results?buildId=471533&view=results

### CODE REVIEW

**Review by mfrw — APPROVED** [review]

File: SPECS/curl/curl.spec [hunk 1]
```diff
@@ -1,15 +1,13 @@
 Summary:        An URL retrieval utility and library
 Name:           curl
-Version:        8.3.0
-Release:        2%{?dist}
+Version:        8.5.0
```
Comment (mfrw) [inline SPECS/curl/curl.spec:3]: Lets try to build the dependents as well before we merge.

Last I recall, there used to be a few `perl-*` packages that complained whenever we did a big jump to curl.

Comment (Camelron) [reply SPECS/curl/curl.spec:3]: To be clear, you mean dash-rolling things that BuildRequires any curl package/subpackage? Should I perform this step for the (already merged) 2.0 PR as well? https://github.com/microsoft/CBL-Mariner/pull/7059

Comment (mfrw) [reply SPECS/curl/curl.spec:3]: No sir .. not a dash-roll, just a sanity check to see if everything builds as it should :)
It most likely would, but just a double check :)

Comment (Camelron) [reply SPECS/curl/curl.spec:3]: I have crafted this spell to get a list of packages to buddybuild.

`grep -Rl 'BuildRequires:.*curl' SPECS/ | xargs -I % bash -c 'basename $(dirname "%")' | tr '\n' ' '`

See the following buddybuild: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=475045&view=results

**Review by mfrw — APPROVED** [review]

**Comment by Camelron** [comment]
So the buddy build failed for a pipeliney-looking reason. I kicked off the standard dev pipelines and got good results for amd64,arm64:

https://dev.azure.com/mariner-org/mariner/_build/results?buildId=475097&view=results
https://dev.azure.com/mariner-org/mariner/_build/results?buildId=475106&view=results

**Pull request merged**